src/services/osc.cpp

```diff
@@ -0,0 +1,128 @@
+#include "osc.hpp"
+#include "tinyosc.h"
+#include "core/globals.hpp"
+#include "services/logger.hpp"
+#include "services/ui.hpp"
+#include "board/ui/keys.hpp"
+
+#include <arpa/inet.h>
+#include <sys/select.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include "tinyosc.h"
+
+namespace otto::service::osc {
+  
+  static volatile bool keepRunning = true;
+
+  // handle Ctrl+C
+  static void sigintHandler(int x) {
+    keepRunning = false;
+  }
+
+  void init()
+  {
+    LOGI("tinyosc init!");
+    
+    char buffer[2048]; // declare a 2Kb buffer to read packet data into
+
+    //// printf("Starting write tests:\n");
+//     int len = 0;
+//     char blob[8] = {0x01, 0x23, 0x45, 0x67, 0x89, 0xAB, 0xCD, 0xEF};
+//     len = tosc_writeMessage(buffer, sizeof(buffer), "/address", "fsibTFNI",
+//         1.0f, "hello world", -1, sizeof(blob), blob);
+//     tosc_printOscBuffer(buffer, len);
+//     printf("done.\n");
+
+    // register the SIGINT handler (Ctrl+C)
+    signal(SIGINT, &sigintHandler);
```

Comment:
yeah, remove this, and run the loop on `otto::global::running()`, that should do the trick